### DISCUSSION TITLE
Why is a space required at the end of a Markdown shortcut?

### DISCUSSION THREAD

**Discussion by ryo-imai-och** [facebook/lexical#4515] [Q&A]
I noticed when customizing the Markdown shortcuts that the last space is mandatory due to the following code included in MarkdownShortcuts.ts:

```typescript
if (textContent[anchorOffset - 1] !== ' ') {
    return false;
}
```

For example, I want to modify it so that it's applied just by inputting `---`, like HorizontalRules, but it doesn't work unless I include a trailing space in the regular expression.

```typescript
// regExp: /^(---|\*\*\*|___)\s?$/,
 regExp: /^(---|\*\*\*|___)?$/,  // <- Not working
```

Of course, if I customize MarkdownShortcuts.ts, it works even without a space. Is there any significant reason for this behavior?

**Comment by RegiByte** [comment]
Have you been able to solve this?

From my perspective, the space is required as a trigger to prevent unwanted transformations from happening when the user is just typing stuff.

Imagine that for some reason a user just wants to put the `---` or `-------` as a separator rather than a full HorizontalRule in some lines of the document.

If you automatically trigger the transformation the user won't be able to do that because the content will be transformed every time as it is being typed out.

That's what I believe, it makes sense for most cases.

**Reply by ryo-imai-och** [reply]
Yes, I think it makes sense in many cases.
Therefore, I understand that the standard "MarkdownTransformers.ts" describes the rules for including trailing spaces.

However, in some cases we may wish to eliminate the need for trailing spaces. In that case, we would have to customize `MarkdownTransformers.ts` to change the rule. However, even if we change it here, the `MarkdownShortcuts.ts` side forces us to require trailing spaces. We would like to know the rational for this.
